### ISSUE TITLE
Always crashes after first cutscene

### ISSUE THREAD

**Issue by GibStorm** [KozGit/DOOM-3-BFG-VR#104]
Just installed DOOM 3 BFG Edition, added the texture pack, installed the mod. Game opens fine, can navigate menus fine etc. When starting the game, the cinematic explaining what the UAC is plays. Just when the next cutscene is supposed to show (with the landing sequence etc) the game just crashes, every time.

**Comment by jamesksu** [comment]
I am also having this same issue.  Crashes right after the first cutscene explaining UAC...

**Comment by jimbobgator** [comment]
Same for me.

**Comment by Andenlol** [comment]
You have to delete the "generated" folder if you have installed the texture pack. It might solve the problem. I'm not an expert, I just read it on reddit.

**Comment by GibStorm** [comment]
I've already deleted the "generated" folder. Doesn't help.

**Comment by Andenlol** [comment]
Oh ok, then I don't know.

**Comment by jimbobgator** [comment]
Yeah, same here. I removed the "generated" folder and the German language file. Thanks for trying to help though.

**Comment by CarlKenner** [comment]
Explaining the UAC isn't a cutscene, it's an FMV that's part of the menu. Please edit the title to crashing BEFORE the first cutscene.

**Comment by jimbobgator** [comment]
I reinstalled this morning without the texture pack and it worked fine. Must be an issue with the texture pack install. I'll try installing it again tonight. Does anyone have a link to the correct texture pack?

I've used the one mentioned here: https://www.reddit.com/r/Vive/comments/5e1cbp/allinone_doom_3_bfg_vr_mod_texture_pack_guide/

**Comment by CarlKenner** [comment]
That is the "correct" texture pack. But I haven't tested it yet myself, so I can't help much. Kozgit (Samson) knows more about it.

**Comment by mogwaimon** [comment]
I am also having the same situation. I am working off a fresh install of Doom 3 BFG edition as well. The game seems to load fine with just the VR mod installed (I used the installer exe to install) but once I add the texture pack from jimbobgator's link the game always crashes on load after the intro FMV.

I install the texture pack either before or after the VR mod, I've tried both ways. I always copy the base folder from the texture pack archives into the root Doom 3 folder, not into the folder with the assets for the VR mod. I've tried both with and without the teleportation AAS files installed for the VR mod. I also have either manually deleted the generated folder inside the texture pack or used the provided .bat file to do so, 

PC specs are i7 4790k, 8GB DDR3 RAM, GTX980 (Strix OC), running on Win 10. The VR HMD I own is the Vive. I have a PS4 controller plugged into one of my USB ports but am attempting to use the Vive controllers.

**Comment by Manzion** [comment]
I had the same problem with the texture pack.
Deleting the folder "C:\Users\<USERNAME>\Saved Games\id Software\DOOM 3 BFG\Fully Possessed" before starting the game worked for me.
Of course you are losing your settings and saved games, so perhaps you should make a backup first.

**Comment by CarlKenner** [comment]
Maybe just try deleting the vr_*.cfg files?

**Comment by mogwaimon** [comment]
I'd tried that before so I tried again just to be safe. This is my first time playing Doom 3 so I didn't really have anything to lose.

The game still crashes. unfortunately. Is there a way to log output to a file for debugging purposes in the Doom 3 engine, and if so where are the logs typically kept?

**Comment by CarlKenner** [comment]
I don't know. All my logs appear in Visual Studio.

I couldn't replicate this problem. So maybe that means it's fixed in the new version, or there's something different about your setup.

**Comment by mogwaimon** [comment]
To clarify, I am running the newest version of Fully Possessed, and I did not have a setup prior to the March 3rd version because I did not have my Vive until last Thursday.

**Comment by mogwaimon** [comment]
Faulting application name: Doom3BFGVR.exe, version: 1.0.0.1, time stamp: 0x58b7ee68
Faulting module name: Doom3BFGVR.exe, version: 1.0.0.1, time stamp: 0x58b7ee68
Exception code: 0xc0000409
Fault offset: 0x00449844
Faulting process id: 0x2668
Faulting application start time: 0x01d298f5634dadb2
Faulting application path: C:\Program Files (x86)\Steam\steamapps\common\DOOM 3 BFG Edition\Doom3BFGVR.exe
Faulting module path: C:\Program Files (x86)\Steam\steamapps\common\DOOM 3 BFG Edition\Doom3BFGVR.exe
Report Id: 0e39c5a4-8abf-4920-9bb8-3a84581eee3f
Faulting package full name: 
Faulting package-relative application ID: 

This is the data I've pulled for one of the crashes from my Event Viewer, I'll see if I can find a logging process for Doom 3 and get some output from the engine itself

**Comment by mogwaimon** [comment]
[qconsole.txt](https://github.com/KozGit/DOOM-3-BFG-VR/files/832001/qconsole.txt)


I've attached (I think) the log file that was put out by Doom 3. The error message seems to point to trying to load source code from the D drive, which is odd in my case because my D drive is the system reserved partition for my OS drive, which shouldn't be accessed by any programs by default.

**Comment by CarlKenner** [comment]
It's not trying to load source from your D drive. The source code was built from my D drive, into the executable. The error is that the game tried to store some text in a fixed length string that didn't have enough room.

**Comment by mogwaimon** [comment]
That makes more sense, I was beginning to think that perhaps the BFG edition used so weird virtual drive trickery to load files (Ports/remakes do some very odd stuff sometimes, see the Steam port of Final Fantasy 6 that pulls info from an actual ROM of the GBA port.)

I'm not much of a coder, but is this something that can be fixed on my end or does it have to be fixed on yours? Hell, is it even under your jurisdiction, so to speak, considering the issue only seems to occur with the high-res texture pack installed?

**Comment by CarlKenner** [comment]
Anyway, there's a new version now for you to try: https://github.com/KozGit/DOOM-3-BFG-VR/pull/110

**Comment by mogwaimon** [comment]
Oh jeez, there was a new update? Sorry I didn't see it, I was using the releases section. Will test out and get back to you in a bit if it's fixed

**Comment by CarlKenner** [comment]
I can't edit the releases section because I don't have write access.
I only posted it a few minutes ago.

**Comment by mogwaimon** [comment]
Oh, my bad. It said 4 days ago on the post so I was surprised. In any case...

I did an uninstall of v20 using the provided uninstall program, deleted the Fully Possessed folder in users/saved games/Doom 3 BFG/etcetera, etcetera, installed v21, and got another crash.

However the log is slightly different, possibly because it's being generated off a fresh install.
[qconsole.txt](https://github.com/KozGit/DOOM-3-BFG-VR/files/832152/qconsole.txt)

Now it's crashing with the same error but after generating assets for the map load. On second boot of Doom3bFGvr.exe the log file is now the same as it was before.

**Comment by KozGit** [comment]
Try deleting ( back up first if necessary ) the entire id software save game directory.  Also, if you can, post a screenshot or directory listing of your doom 3 directory tree, with the base directory expanded.

The game is crashing processing a string while preloading images.  The only strings involved are the resource names - maybe something is wrong with your hidef texture pack or the way it's installed.

**Comment by mogwaimon** [comment]
![doom3bfgfilestructure](https://cloud.githubusercontent.com/assets/25550869/23883887/16390156-0840-11e7-87b9-4d0caafcda50.jpg)


Here's the screenshot. Left side is the base folder of Doom 3 BFG. Upper right is where I have the texture pack extracted, it is intact except for the generated folder which I'd already removed prior to copy/pasting into the Doom 3 BFG base folder. Lower right is most of the root folder of the Doom 3 BFG Edition.

If you reply I'll likely get back to you tomorrow at the earliest, just got home from work and I'm too beat for troubleshooting tonight

**Comment by KozGit** [comment]
I'm glad you got this working! Would you mind providing what the full original path name was?  From the picture, it appears to match mine almost exactly.  There is a 260 character limit for path names in explorer,  but from what I can tell from the pic you provided, even with the 'Fully Possessed' folder added to the steam game directory, it doesn't look like it should exceed the limit, which is curious.

**Comment by mogwaimon** [comment]
I use 7-zip for all my archival needs, have been for years. My Win10 install is fairly recent, I think it was maybe December or January that I finally updated so my 7-zip should be the latest stable release

The full original path should be C:\Program Files (x86)\Steam\steamapps\common\Doom 3 BFG Edition, but you've got to remember all the various sub-folders as well as the length of some of these image names. As an example, one of the deeper nested images would be something like C:\Program Files (x86)\Steam\steamapps\common\Doom 3 BFG Edition\Fully Possessed\generated\images\makealpha\models\items\flashlight\beam1#__0200.bimage, so as you can see those characters really add up quick

**Comment by KozGit** [comment]
I'll  look at this a little more.  Just did a little testing, and if the path to the Doom 3 BFG Edition directory is more than 22 characters longer than the default of  c:\Program Files (x86)/Steam/steamapps/common/Doom 3 BFG Edition, the game will crash. 

Edit:
A quick test just revealed that images created by the addnormal function in a material can be given ridiculously long filenames ( Image 2315 name 'addnormals( models/characters/male_npc/marine/marine_arm2_local, heightmap( models/characters/male_npc/marine/marine_arm2_h, 5))', which don't seem to be length checked when the binary image is being created, which is what appears to be causing this issue.

**Comment by mogwaimon** [comment]
Good, good. I'm glad that there might be a solution in the code; or at least something to help mitigate future occurrences. Thanks for your help man, can't wait to get into the game finally when I have more time

**Comment by CarlKenner** [comment]
The 260 character limit can be fixed if you add "\\?\" to the start of the path, and use W (unicode) versions of the APIs.

**Comment by CarlKenner** [comment]
OK, I fixed the 260 character limit for most things (I think).
